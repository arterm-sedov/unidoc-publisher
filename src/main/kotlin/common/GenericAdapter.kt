package common

import model.Node
import reader.HtmlNode
import reader.UnknownTagProcessing
import writer.OdtStyleList

interface GenericAdapter {
    val unkonwnTagProcessingRule: HtmlNode.() -> UnknownTagProcessing
    fun basicStyle(): OdtStyleList
    fun Node.normalizeAll() {}
}