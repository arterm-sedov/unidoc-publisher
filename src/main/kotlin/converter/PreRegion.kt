package converter

data class PreRegion(
    val includeTags: Set<String>,
    val pre: String,
    val inLine: Boolean
)
