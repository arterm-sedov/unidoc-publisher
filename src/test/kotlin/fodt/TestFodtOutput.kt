package fodt

import common.prettySerialize
import converter.PreRegion
import org.approvaltests.Approvals
import org.intellij.lang.annotations.Language
import org.junit.jupiter.api.Test
import org.w3c.dom.Document
import org.xml.sax.InputSource
import java.io.StringReader
import java.io.StringWriter
import javax.xml.parsers.DocumentBuilderFactory
import javax.xml.transform.OutputKeys
import javax.xml.transform.TransformerFactory
import javax.xml.transform.dom.DOMSource
import javax.xml.transform.stream.StreamResult

class TestFodtOutput {
    @Test
    fun testSimpleFodtOutput() {
        val preFodt =
            """
                <root xmlns:text="urn:oasis:names:tc:opendocument:xmlns:text:1.0"
                      xmlns:style="urn:oasis:names:tc:opendocument:xmlns:style:1.0"
                      xmlns:fo="urn:oasis:names:tc:opendocument:xmlns:xsl-fo-compatible:1.0">
                    <text:h text:style-name="Heading 1">
                        <style:text-properties fo:font-size="14pt"/>
                        Heading 1
                    </text:h>
                    <text:p text:style-name="Body Text">
                        Some text
                    </text:p>
                    <text:h text:style-name="Heading 1">
                        <style:text-properties fo:font-size="14pt"/>
                        Heading 1
                    </text:h>
                </root>
        """

        val template =
            """
                <office:document
                        xmlns:office="urn:oasis:names:tc:opendocument:xmlns:office:1.0"
                        xmlns:text="urn:oasis:names:tc:opendocument:xmlns:text:1.0"
                        xmlns:style="urn:oasis:names:tc:opendocument:xmlns:style:1.0"
                        xmlns:fo="urn:oasis:names:tc:opendocument:xmlns:xsl-fo-compatible:1.0">
                    <office:automatic-styles>
                        <this-tag-should-be-left/>
                    </office:automatic-styles>
                    <office:body>
                        <office:text text:use-soft-page-breaks="true">
                            <this-tag-should-be-left/>
                            <text:p><text:variable-set text:name="include" office:value-type="string">all</text:variable-set></text:p>
                            <this-tag-should-be-left/>
                        </office:text>
                    </office:body>
                </office:document>
            """

        val preFodtList = arrayListOf<PreRegion>(
            PreRegion(setOf("all"), preFodt, false)
        )

        FodtGenerator(preFodtList, template)
            .enrichedTemplate
            .prettySerialize()
            .apply {
                Approvals.verify(this)
            }
    }

}

