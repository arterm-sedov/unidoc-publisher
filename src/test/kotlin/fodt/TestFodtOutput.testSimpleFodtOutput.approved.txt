<office:document>  
  <office:automatic-styles> 
    <this-tag-should-be-left/>  
    <style:style
                    style:family="paragraph"
                    style:name="PAMC1"
                    style:parent-style-name="Heading_20_1">
      <style:text-properties
                    fo:font-size="14pt"/>
    </style:style>
    <style:style
                    style:family="paragraph"
                    style:name="PAMC2"
                    style:parent-style-name="Body_20_Text"/>
    <style:style
                    style:family="paragraph"
                    style:name="PAMC3"
                    style:parent-style-name="Heading_20_1">
      <style:text-properties
                    fo:font-size="14pt"/>
    </style:style>
  </office:automatic-styles>  
  <office:body> 
    <office:text
                    text:use-soft-page-breaks="true"> 
      <this-tag-should-be-left/>  
      <text:h
                    text:style-name="PAMC1">Heading 1</text:h>
      <text:p
                    text:style-name="PAMC2">Some text</text:p>
      <text:h
                    text:style-name="PAMC3">Heading 1</text:h>  
      <this-tag-should-be-left/> 
    </office:text> 
  </office:body> 
</office:document>
